Address some of the Safer CPP warnings in WKRemoteObjectCoder.mm

cdumez opened the pull request:
#### be4db26277ae5625c5ac485d25a5cadf3e67ce7e
<pre>
Address some of the Safer CPP warnings in WKRemoteObjectCoder.mm
<a href="https://bugs.webkit.org/show_bug.cgi?id=287465">https://bugs.webkit.org/show_bug.cgi?id=287465</a>

Reviewed by Darin Adler.

* Source/WebKit/Shared/API/Cocoa/WKRemoteObjectCoder.mm:
(-[WKRemoteObjectEncoder init]):
(encodeToObjectStream):
(encodeString):
(encodeObject):
(-[WKRemoteObjectEncoder encodeObject:forKey:]):
(-[WKRemoteObjectEncoder encodeBytes:length:forKey:]):
(-[WKRemoteObjectEncoder encodeBool:forKey:]):
(-[WKRemoteObjectEncoder encodeInt:forKey:]):
(-[WKRemoteObjectEncoder encodeInt32:forKey:]):
(-[WKRemoteObjectEncoder encodeInt64:forKey:]):
(-[WKRemoteObjectEncoder encodeFloat:forKey:]):
(-[WKRemoteObjectEncoder encodeDouble:forKey:]):
(-[WKRemoteObjectDecoder initWithInterface:rootObjectDictionary:replyToSelector:]):
(-[WKRemoteObjectDecoder containsValueForKey:]):
(decodeString):
(decodeObject):
(-[WKRemoteObjectDecoder decodeBoolForKey:]):
(-[WKRemoteObjectDecoder decodeIntForKey:]):
(-[WKRemoteObjectDecoder decodeInt32ForKey:]):
(-[WKRemoteObjectDecoder decodeInt64ForKey:]):
(-[WKRemoteObjectDecoder decodeFloatForKey:]):
(-[WKRemoteObjectDecoder decodeDoubleForKey:]):
(-[WKRemoteObjectDecoder decodeBytesForKey:returnedLength:]):
(-[WKRemoteObjectDecoder decodeObjectOfClasses:forKey:]):

Canonical link: <a href="https://commits.webkit.org/290294@main">https://commits.webkit.org/290294@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4fa7507bbc1e17e2c085ac772278c89162fb99a9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/89445 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/8983 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/44300 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/94443 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/40212 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/9365 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/17257 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/68911 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/26571 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/92447 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/7186 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/81186 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/49273 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/6923 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/39327 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/77281 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/36569 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/96271 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/16633 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/12226 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/77782 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/16889 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/76986 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/77094 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/19048 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/21527 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/20106 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/9778 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/16646 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/21950 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/16387 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/19838 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/18169 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebKit/Shared/API/Cocoa/WKRemoteObjectCoder.mm:
```diff
@@ -91,10 +92,10 @@ - (BOOL)validateClassSupportsSecureCoding:(Class)objectClass;
 @implementation WKRemoteObjectEncoder {
-    RefPtr<API::Dictionary> _rootDictionary;
-    API::Array* _objectStream;
+    const RefPtr<API::Dictionary> _rootDictionary;
```
I’m surprised this works in an Objective-C object. It’s sort of amazing that `lazyInitialize` is powerful enough to do this even after an early return from the `init` method. Also slightly surprised it’s a `RefPtr` and not a `Ref`.

cdumez replied:
Ref wouldn't work with Objective C and init functions because Ref doesn't have a default constructor.

cdumez committed:
Address some of the Safer CPP warnings in WKRemoteObjectCoder.mm
https://bugs.webkit.org/show_bug.cgi?id=287465

Reviewed by Darin Adler.

* Source/WebKit/Shared/API/Cocoa/WKRemoteObjectCoder.mm:
(-[WKRemoteObjectEncoder init]):
(encodeToObjectStream):
(encodeString):
(encodeObject):
(-[WKRemoteObjectEncoder encodeObject:forKey:]):
(-[WKRemoteObjectEncoder encodeBytes:length:forKey:]):
(-[WKRemoteObjectEncoder encodeBool:forKey:]):
(-[WKRemoteObjectEncoder encodeInt:forKey:]):
(-[WKRemoteObjectEncoder encodeInt32:forKey:]):
(-[WKRemoteObjectEncoder encodeInt64:forKey:]):
(-[WKRemoteObjectEncoder encodeFloat:forKey:]):
(-[WKRemoteObjectEncoder encodeDouble:forKey:]):
(-[WKRemoteObjectDecoder initWithInterface:rootObjectDictionary:replyToSelector:]):
(-[WKRemoteObjectDecoder containsValueForKey:]):
(decodeString):
(decodeObject):
(-[WKRemoteObjectDecoder decodeBoolForKey:]):
(-[WKRemoteObjectDecoder decodeIntForKey:]):
(-[WKRemoteObjectDecoder decodeInt32ForKey:]):
(-[WKRemoteObjectDecoder decodeInt64ForKey:]):
(-[WKRemoteObjectDecoder decodeFloatForKey:]):
(-[WKRemoteObjectDecoder decodeDoubleForKey:]):
(-[WKRemoteObjectDecoder decodeBytesForKey:returnedLength:]):
(-[WKRemoteObjectDecoder decodeObjectOfClasses:forKey:]):

Canonical link: https://commits.webkit.org/290294@main
